MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Toolbar/LanguagePicker.swift

```diff
@@ -36,31 +75,42 @@ struct LanguagePicker: View {
                     }
                     return Text("")
                 }()
-                Button(action: { onSelect(lang.id) }) {
-                    ViewThatFits(in: .horizontal) {
-                        HStack(spacing: 0) { endonym; Text(" "); exonym }
-                        VStack(alignment: .leading) { endonym; exonym }
+                Button(action: {
+                    selectedLanguage = lang.id
+                    onSelect(lang.id)
+                }) {
+                    HStack {
+                        ViewThatFits(in: .horizontal) {
+                            HStack(spacing: 0) { endonym; Text(" "); exonym }
+                            VStack(alignment: .leading) { endonym; exonym }
+                        }
+                        if lang.id == selectedLanguage {
+                        Spacer()
+                          Image(systemName: "checkmark")
+                                .foregroundStyle(Asset.Colors.Brand.blurple.swiftUIColor)
+                        }
                     }
                 }
                 .tint(.primary)
                 .accessibilityLabel(Text(lang.label))
-            }.toolbar {
-                ToolbarItem(placement: .cancellationAction) {
-                    Button(L10n.Common.Controls.Actions.cancel) {
-                        dismiss()
-                    }
-                }
+                .id(lang.id)
             }
             .listStyle(.plain)
             .searchable(text: $query, placement: .navigationBarDrawer(displayMode: .always))
-            .navigationTitle(L10n.Scene.Compose.Language.title)
-            .navigationBarTitleDisplayMode(.inline)
-        }.navigationViewStyle(.stack)
+            .onAppear {
+                filteredLanguages = query.isEmpty ? languages : languages.filter { $0.contains(query) }
+                DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(200)) { // when scrolling to quickly it'll overlap with other drawcycles and mess up the position :-(
```

Comment:
To it felt weird that the search bar appears without an animation. But I don't know how to improve this. Do you have an idea? (Feel free to just mark this as resolved)

https://github.com/mastodon/mastodon-ios/assets/2580019/1e0f4441-94b0-467b-a45d-568092c0eca8

Reply:
I've tried a lot of things, but unfortunately not found any good solution to fix this. It's due to the `.searchable(:)` viewModifier used in `LanguagePicker`.

Reply:
I'll merge as-is for now.